core/unit_test/TestExecutionSpace.hpp

```diff
@@ -50,4 +50,61 @@ TEST(TEST_CATEGORY, execution_space_as_class_data_member) {
 }
 #endif
 
+template <class ExecutionSpace>
+struct CheckExecutionSpaceStatus {
+  KOKKOS_FUNCTION void operator()(int i, int& update) const { update += i; };
+
+  CheckExecutionSpaceStatus() { check(); }
+
+  void check() const {
+    ExecutionSpace exec;
+    ASSERT_EQ(exec.is_running(), false);
+
+    Kokkos::View<int, typename ExecutionSpace::memory_space> result_view(
+        "result_view");
+    int result;
+
+    Kokkos::deep_copy(exec, result, result_view);
+    while (exec.is_running() == true) {
+    }
+    ASSERT_EQ(result, 0);
+
+    Kokkos::deep_copy(result_view, 1);
+    ASSERT_EQ(exec.is_running(), false);
+    Kokkos::deep_copy(exec, result, result_view);
+    exec.fence();
+    ASSERT_EQ(exec.is_running(), false);
+    ASSERT_EQ(result, 1);
+
+// FIXME OPENACC
+#ifdef KOKKOS_ENABLE_OPENACC
+    if constexpr (std::is_same_v<ExecutionSpace,
+                                 Kokkos::Experimental::OpenACC>) {
+      GTEST_SKIP() << "skip the other half of the test because of missing "
+                      "functionality in OpenACC backend";
+    }
+#endif
+
+    const int N = 10000;
+
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(exec, 0, N),
+                            *this, result_view);
+    Kokkos::deep_copy(exec, result, result_view);
+    while (exec.is_running() == true) {
```

Comment:
```suggestion
    while (exec.is_running()) {
```

Reply:
Per dev meeting add a comment that this happens to work for now but that we don't want users to write this code and we reserve the right to have a lazy implementation which would make this hang.